adoc/architecture-updates.adoc

```diff
@@ -0,0 +1,146 @@
+ag-master-caasp-2   SLE 15 SP1   4.12.14-1p150.12.28-default   docker://18.6.1     yes           yes
+ag-worker-caasp-0   SLE 15 SP1   4.12.14-1p150.12.28-default   docker://18.6.1     yes           yes
+----
+
+==== Node reboots
+
+Some updates require the node to be rebooted to make them active. As a system administrator
+I don’t have to worry about a thing.
+
+The {productname} cluster is configured by default to take advantage of ​kured​. This service look for
+nodes that have to be rebooted and, before doing the actual reboot, takes care of draining the
+node.
+
+Kured reboots one node per time. This ensures my worker nodes won’t be saturated and my
+etcd cluster will always be in a healthy state.
+
+System administrators can integrate kured stats into my prometheus instance and create alerts,
+charts and other personal customizations.
+
+System administrators can also fine tune the kured deployment to prevent its agent from
+rebooting machines where special workloads are running. For example: I can prevent kured
+from rebooting nodes running computational workloads until their pods are done. To achieve
+that I just have to instruct all my users to add special labels to the pods they don’t want to see
+interrupted due to the node being rebooted.
+
+==== Interactive upgrades
+
+Some updates might cause damages to the running workloads. For example, docker updates
+can cause all the containers running on a machine to be suddenly killed.
+Note well:​ these interactive updates are currently being referenced by this document as
+“disruptive upgrades”.
+
+System administrators don’t have to worry about disruptive upgrades. {productname} will
+automatically apply them making sure no disruption is caused to the cluster.
+That happens because nodes with disruptive upgrades are updated one at a time, a bit like
+when nodes are automatically rebooted. Moreover {productname} will take care of cordoning and
+uncordoning the node before these updates are applied.
+
+Disruptive upgrades could take some time to be automatically applied due to their sensitive
+nature (nodes are updated one by one). As a system operator I can always see the status of my
+nodes by looking at the annotations of the kubernetes nodes (see previous section).
+
+By looking at node annotations a system administrator can answer the following questions:
+
+* Does the node have pending disruptive upgrades?
+* Are the disruptive upgrades being applied?
+
+=== Upgrades of OS components (not automated)
+
+As a systems administrator I want my cluster to be updated only when I consider it’s necessary,
+with human intervention as a must for every package update, so that I’m in complete control of
+when my cluster is updated.
+By default {productname} clusters have some updates applied automatically. Nodes can also be
+rebooted in an automatic fashion under some circumstances.
+To prevent that from happening I have to disable a systemd timer on each node of the cluster.
```

Comment:
It's also possible to annotate nodes so kured won't automatically reboot the node (even leaving the systemd timer running), see: https://github.com/weaveworks/kured#disabling-reboots

This has the advantage that a user that wants to temporarily or permanentely disable reboots don't need SSH access to the machine, they can annotate the node using `kubectl` if their RBAC policies allow the user to do so.